runtime/io/jvm/src/aws/smithy/kotlin/runtime/io/JavaIO.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.io
+
+import aws.smithy.kotlin.runtime.io.internal.toSdk
+import java.io.File
+import java.io.InputStream
+import java.io.OutputStream
+import java.nio.file.Path
+import okio.sink as okioSink
+import okio.source as okioSource
+
+/**
+ * Create a [SdkSource] from the given path and range
+ */
+public fun Path.source(range: LongRange): SdkSource = source(range.first, range.last)
+
+/**
+ * Create a [SdkSource] from the given path and range
+ */
+public fun Path.source(start: Long = 0L, endInclusive: Long = -1): SdkSource {
+    val f = toFile()
+    require(f.exists()) { "cannot create SdkSource, file does not exist: $this" }
+    require(f.isFile) { "cannot create a SdkSource from a directory: $this" }
```

Comment:
**Nit**: These checks probably belong in the `File` variant of the method instead.